bgpd/bgp_updgrp_adv.c

```diff
@@ -991,18 +993,21 @@ void subgroup_default_originate(struct update_subgroup *subgrp, bool withdraw)
 		if (dest) {
 			for (pi = bgp_dest_get_bgp_path_info(dest); pi;
 			     pi = pi->next) {
-				if (CHECK_FLAG(pi->flags, BGP_PATH_SELECTED))
+				if (CHECK_FLAG(pi->flags, BGP_PATH_SELECTED)) {
 					if (subgroup_announce_check(
 						    dest, pi, subgrp,
 						    bgp_dest_get_prefix(dest),
 						    &attr, NULL)) {
 						struct attr *default_attr =
 							bgp_attr_intern(&attr);
 
-						bgp_adj_out_set_subgroup(
-							dest, subgrp,
-							default_attr, pi);
-					}
+						if (!bgp_adj_out_set_subgroup(
+							    dest, subgrp,
+							    default_attr, pi))
+							bgp_attr_flush(&attr);
```

Comment:
Looks `bgp_attr_flush(&attr)` is not reasonable. Should it be `bgp_attr_unintern(&default_attr)` regarding the using of `bgp_attr_intern(&attr)` at line 1002?

Reply:
> Looks `bgp_attr_flush(&attr)` is not reasonable. Should it be `bgp_attr_unintern(&default_attr)` regarding the using of `bgp_attr_intern(&attr)` at line 1002?

Suppose whatever `bgp_adj_out_set_subgroup` returns true or false, always need to call `bgp_attr_unintern(&default_attr)`

Reply:
In such a case `bgp_attr_unintern(&default_attr)` will lead to another memory leak where attributes under `attr` are modified by `subgroup_announce_check()`, I think...

Reply:
I think a flush is not going to do anything here, but I also think that an unintern is the appropriate thing to do here.  If there is a failure to send this prefix at this point in time.  FRR should come back around and resend the route again at some point in time in the future.  In which case the intern would just return the saved data.  So it's ok in my opinion.

Reply:
hi @donaldsharp  can I understand that the change would be from the bgp_attr_flush to bgp_attr_unintern at line 1007?

Reply:
no that is not what I said.  I don't think it is necessary.  Nothing will be leaked then and when the default originate can be sent again it will be ok as that the data will be there when it is interned again.